File: backend/app/api/api_v1/endpoints/feedback.py
from typing import List
from fastapi import APIRouter, Depends, HTTPException
from sqlalchemy.ext.asyncio import AsyncSession

from app.db.base import get_db
from app.schemas.feedback import FeedbackMessage, FeedbackMessageCreate, FeedbackMessageUpdate
from app.crud.feedback import feedback as crud_feedback

router = APIRouter()


@router.post("/", response_model=FeedbackMessage)
async def create_feedback(
    feedback_in: FeedbackMessageCreate,
    db: AsyncSession = Depends(get_db)
):
    """Create a new feedback message"""
    return await crud_feedback.create(db=db, obj_in=feedback_in)


@router.get("/", response_model=List[FeedbackMessage])
async def read_feedbacks(
    skip: int = 0,
    limit: int = 100,
    db: AsyncSession = Depends(get_db)
):
    """Retrieve feedback messages"""
    return await crud_feedback.get_multi(db, skip=skip, limit=limit)


@router.get("/{feedback_id}", response_model=FeedbackMessage)
async def read_feedback(
    feedback_id: int,
    db: AsyncSession = Depends(get_db)
):
    """Get feedback message by ID"""
    feedback = await crud_feedback.get(db=db, id=feedback_id)
    if not feedback:
        raise HTTPException(status_code=404, detail="Feedback not found")
    return feedback


@router.put("/{feedback_id}", response_model=FeedbackMessage)
async def update_feedback(
    feedback_id: int,
    feedback_in: FeedbackMessageUpdate,
    db: AsyncSession = Depends(get_db)
):
    """Update feedback message"""
    feedback = await crud_feedback.get(db=db, id=feedback_id)
    if not feedback:
        raise HTTPException(status_code=404, detail="Feedback not found")
    return await crud_feedback.update(db=db, db_obj=feedback, obj_in=feedback_in)


@router.delete("/{feedback_id}")
async def delete_feedback(
    feedback_id: int,
    db: AsyncSession = Depends(get_db)
):
    """Delete feedback message"""
    feedback = await crud_feedback.get(db=db, id=feedback_id)
    if not feedback:
        raise HTTPException(status_code=404, detail="Feedback not found")
    await crud_feedback.remove(db=db, id=feedback_id)
    return {"message": "Feedback deleted successfully"}

File: frontend/src/services/feedback.js
import api from './api'

export const feedbackService = {
  async createFeedback(feedbackData) {
    try {
      const response = await api.post('/feedback/', feedbackData)
      return response.data
    } catch (error) {
      throw error.response?.data || error.message
    }
  },

  async getFeedbacks() {
    try {
      const response = await api.get('/feedback/')
      return response.data
    } catch (error) {
      throw error.response?.data || error.message
    }
  },

  async getFeedback(id) {
    try {
      const response = await api.get(`/feedback/${id}`)
      return response.data
    } catch (error) {
      throw error.response?.data || error.message
    }
  },

  async updateFeedback(id, feedbackData) {
    try {
      const response = await api.put(`/feedback/${id}`, feedbackData)
      return response.data
    } catch (error) {
      throw error.response?.data || error.message
    }
  },

  async deleteFeedback(id) {
    try {
      const response = await api.delete(`/feedback/${id}`)
      return response.data
    } catch (error) {
      throw error.response?.data || error.message
    }
  }
}